cfme/storage/volume.py

```diff
@@ -121,22 +167,88 @@ def is_displayed(self):
     form = View.nested(VolumeAddForm)
 
 
+class VolumeCollection(Navigatable):
+    """Collection object for the :py:class:'cfme.storage.volume.Volume'. """
+
+    def instantiate(self, name, provider):
+        return Volume(name, provider, collection=self)
+
+    def delete(self, *volumes):
+        """Delete one or more Volumes from list of Volumes
+
+        Args:
+            One or Multiple 'cfme.storage.volume.Volume' objects
+        """
+
+        view = navigate_to(self, 'All')
+
+        if view.entities.get_all():
```

Comment:
this will work well only if all volumes on the same page

Reply:
Yes on **Volume All** page, we will get all volumes to check volume available for iteration or no volume available for deletion purpose.

Reply:
I mean this method will fail if some of passed volumes are on another page not current one